packages/core/src/middleware/koa-cors.ts

```diff
@@ -10,12 +11,21 @@ export default function koaCors<StateT, ContextT, ResponseBodyT>(
     origin: (ctx) => {
       const { origin } = ctx.request.headers;
 
-      return origin &&
-        urlSets
-          .flatMap((set) => set.deduplicated())
-          .some((value) => new URL(value).origin === origin)
-        ? origin
-        : '';
+      if (
+        origin &&
+        urlSets.some((set) =>
+          set.deduplicated().some(
+            (url) =>
+              url.origin === origin &&
+              // Disable localhost CORS in production since it's unsafe
+              !(EnvSet.values.isProduction && url.hostname === 'localhost')
```

Comment:
This doesn't seem right to me. Correct me if I'm wrong, but I think we should always allow CORS for localhost? And this code does not seem to allow CORS for localhost